### ISSUE TITLE
crash after rollback to savepoint

### ISSUE THREAD

**Issue by pykello** [citusdata/citus#3622]
On master, I did this:

```sql

create table t(a int);
select create_distributed_table('t', 'a');

begin;
insert into t values (4);
savepoint s1;
insert into t select i from generate_series(1, 10000000) i;
(wait for 2 seconds and then cancel)
rollback to savepoint s1;
insert into t select i from generate_series(1, 10000000) i;
server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
Time: 1199.724 ms (00:01.200)

```

backtrace was:

```
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007f30599c7801 in __GI_abort () at abort.c:79
#2  0x00005641de0258b3 in ExceptionalCondition (
    conditionName=conditionName@entry=0x7f30569aed30 "!(!placementConnection->hadDML)", 
    errorType=errorType@entry=0x7f30569a7c50 "FailedAssertion", 
    fileName=fileName@entry=0x7f30569aebc8 "connection/placement_connection.c", lineNumber=lineNumber@entry=405) at assert.c:54
#3  0x00007f3056932439 in AssignPlacementListToConnection (placementAccessList=placementAccessList@entry=0x5641e0bdf028, 
    connection=connection@entry=0x5641e0d6bc28) at connection/placement_connection.c:405
#4  0x00007f30569324f6 in StartPlacementListConnection (flags=12, placementAccessList=0x5641e0bdf028, 
    userName=userName@entry=0x5641e04e3c68 "hadi") at connection/placement_connection.c:322
#5  0x00007f30569325e4 in StartPlacementConnection (flags=<optimized out>, placement=placement@entry=0x5641e0b5d9c8, 
    userName=userName@entry=0x5641e04e3c68 "hadi") at connection/placement_connection.c:257
#6  0x00007f3056932606 in GetPlacementConnection (flags=<optimized out>, placement=placement@entry=0x5641e0b5d9c8, 
    userName=userName@entry=0x5641e04e3c68 "hadi") at connection/placement_connection.c:210
#7  0x00007f30569270a0 in CopyGetPlacementConnection (stopOnFailure=false, placement=0x5641e0b5d9c8)
    at commands/multi_copy.c:3228
#8  InitializeCopyShardState (stopOnFailure=false, shardId=<optimized out>, connectionStateHash=0x5641e07c8618, 
    shardState=0x5641e0794fd8) at commands/multi_copy.c:3138
#9  GetShardState (found=0x7ffdf14c9bf2, stopOnFailure=false, connectionStateHash=0x5641e07c8618, 
    shardStateHash=<optimized out>, shardId=<optimized out>) at commands/multi_copy.c:3096
#10 CitusSendTupleToPlacements (copyDest=0x5641e0bdecf8, slot=0x5641e052b3e0) at commands/multi_copy.c:2228
#11 CitusCopyDestReceiverReceive (slot=0x5641e052b3e0, dest=0x5641e0bdecf8) at commands/multi_copy.c:2175
#12 0x00005641ddd88248 in ExecutePlan (execute_once=<optimized out>, dest=0x5641e0bdecf8, direction=-572907836, numberTuples=0, 
    sendTuples=<optimized out>, operation=CMD_SELECT, use_parallel_mode=<optimized out>, planstate=0x5641e052a070, 
    estate=0x5641e0529e18) at execMain.c:1677
#13 standard_ExecutorRun (queryDesc=queryDesc@entry=0x5641e0388878, direction=direction@entry=ForwardScanDirection, 
    count=count@entry=0, execute_once=execute_once@entry=true) at execMain.c:364
#14 0x00007f305694e7eb in CitusExecutorRun (queryDesc=0x5641e0388878, direction=ForwardScanDirection, count=0, 
    execute_once=<optimized out>) at executor/multi_executor.c:220
#15 0x00005641ddefdd05 in PortalRunSelect (portal=portal@entry=0x5641e0440890, forward=forward@entry=true, count=0, 
    count@entry=9223372036854775807, dest=dest@entry=0x5641e0bdecf8) at pquery.c:929
#16 0x00005641ddeff496 in PortalRun (portal=portal@entry=0x5641e0440890, count=count@entry=9223372036854775807, 
    isTopLevel=isTopLevel@entry=false, run_once=run_once@entry=true, dest=dest@entry=0x5641e0bdecf8, 
    altdest=altdest@entry=0x5641e0bdecf8, completionTag=0x0) at pquery.c:770
#17 0x00007f305694f2e4 in ExecutePlanIntoDestReceiver (queryPlan=0x5641e0bdeb10, params=0x0, dest=0x5641e0bdecf8)
    at executor/multi_executor.c:687
#18 0x00007f305694b806 in ExecutePlanIntoRelation (executorState=0x5641e0513ef8, selectPlan=0x5641e0bdeb10, 
    insertTargetList=0x5641e04e21c0, targetRelationId=16814) at executor/insert_select_executor.c:633
#19 CoordinatorInsertSelectExecScanInternal (node=0x5641e0514180) at executor/insert_select_executor.c:361
#20 CoordinatorInsertSelectExecScan (node=0x5641e0514180) at executor/insert_select_executor.c:99
#21 0x00005641ddd8821a in ExecProcNode (node=0x5641e0514180) at ../../../src/include/executor/executor.h:239
#22 ExecutePlan (execute_once=<optimized out>, dest=0x5641e052e9d0, direction=NoMovementScanDirection, numberTuples=0, 
    sendTuples=<optimized out>, operation=CMD_INSERT, use_parallel_mode=<optimized out>, planstate=0x5641e0514180, 
    estate=0x5641e0513ef8) at execMain.c:1646
```

**Comment by serprex** [comment]
This seems to be fixed by #3403

**Comment by pykello** [comment]
> This seems to be fixed by #3403

I tried that branch, and bug is still there. Note that the case I mentioned doesn't use the adaptive executor.
